### PR TITLE
Replace old shared project with real one - Avalonia.PlatformSupport

### PR DESCRIPTION
**Pull request by maxkatz6**
## What does the pull request do?

Removed old shared project and replaces with normal C# project.
Removed code duplication between platform backends.

## Breaking changes

- Binary breacking change (types moved to another assembly)
- Namespace change breaking change (AssetsLoader impl moved to another namespace)

Note, new project might be merged into Avalonia project with [this PR](https://github.com/AvaloniaUI/Avalonia/pull/5831). Otherwise we can consider Numerge-ing some more dlls.

### CODE REVIEW

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018310-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018312-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

**Review by grokys — APPROVED** [review]

**Comment by avaloniaui-team** [comment]
You can test this PR using the following package version. `0.10.999-cibuild0018329-beta`. (feed url: https://nuget.avaloniaui.net/repository/avalonia-all/index.json) [PRBUILDID]

File: src/Avalonia.PlatformSupport/StandardRuntimePlatform.cs [hunk 1]
```diff
@@ -169,5 +165,54 @@ void Free(IntPtr ptr, int len)
...
+            else
+                throw new Exception("Unknown OS platform " + RuntimeInformation.OSDescription);
+#endif
+
+            return new RuntimePlatformInfo
+            {
+#if NETCOREAPP
+                IsCoreClr = true,
+#elif NETFRAMEWORK
+                IsDotNetFramework = false,
```
Comment (DyachenkoA) [inline src/Avalonia.PlatformSupport/StandardRuntimePlatform.cs:204]: IsDotNetFramework = true ?

Comment (maxkatz6) [reply src/Avalonia.PlatformSupport/StandardRuntimePlatform.cs:204]: Thanks. Will fix it in another pr later.

**Pull request merged**